pymc3/distributions/distribution.py

```diff
@@ -328,26 +328,17 @@ def _distr_parameters_for_repr(self):
 class Discrete(Distribution):
     """Base class for discrete distributions"""
 
-    def __init__(self, shape=(), dtype=None, defaults=("mode",), *args, **kwargs):
-        if dtype is None:
-            if aesara.config.floatX == "float32":
-                dtype = "int16"
-            else:
-                dtype = "int64"
-        if dtype != "int16" and dtype != "int64":
-            raise TypeError("Discrete classes expect dtype to be int16 or int64.")
+    def __new__(cls, name, *args, **kwargs):
 
-        super().__init__(shape, dtype, defaults=defaults, *args, **kwargs)
+        if kwargs.get("transform", None):
+            raise ValueError("Transformations for discrete distributions")
+
```

Comment:
Shouldn't we keep the `dtype` checks? (Based on `intX`.)

Reply:
Those are done at the Aesara `Op`-level now (i.e. within `RandomVariable.make_node`); although I'm not sure if float-to-int conversion is part of that.  It might only raise an exception for the wrong dtype.